pyannote/audio/core/task.py

```diff
@@ -221,8 +290,17 @@ def __init__(
         self.protocol, checks = check_protocol(protocol)
         self.has_validation = checks["has_validation"]
         self.has_scope = checks["has_scope"]
+        if not self.has_scope:
+            raise ValueError(
+                "Protocol must provide 'scope' information (e.g. 'file', 'database', or 'global')."
+            )
+
         self.has_classes = checks["has_classes"]
 
+        # metadata cache
+        self.cache = Path(cache) if cache else Path(mktemp())
+        self.prepared_data = {}
```

Comment:
Is this line needed?

Reply:
I don't think it is still needed. 
Can you double-check?

Reply:
I confirm that line `self.prepared_data = {}` is not needed anymore. However, we still need the line `self.cache = ...`, to communicate cache between task initialization, `prepare_data()` and `setup()`.